secant/Features/Send/Views/SendView.swift

```diff
@@ -0,0 +1,58 @@
+import SwiftUI
+import ComposableArchitecture
+
+struct SendView: View {
+    enum Route: Equatable {
+        case showApprove
+        case showSent
+        case done
+    }
+
+    let store: Store<SendState, SendAction>
+
+    var body: some View {
+        WithViewStore(store) { viewStore in
+            Create(
+                transaction: viewStore.bindingForTransaction,
+                isComplete: viewStore.bindingForApprove
+            )
+            .navigationLinkEmpty(
```

Comment:
As we sort of discussed offline, this seems like one of those judgment calls.   I don't think all nesting is bad here, and it certainly could be the cleanest implementation, but having ways to avoid this getting out of hand will probably become important.

Reply:
The main hurdle I had with Navigation is that NavigationLinks that are tied to components that might end up being disabled by state changes, start to corrupt the navigation stack.
For example. The Home screen condenses most of the state. In the reference wallet we had screens that were not accessible while syncing. those were the send screen and the wallet history screen. Innocently (and following Apple's initial documentation) one would do something like

```` swift
NavigationLink(destination: WalletHistory()) { 
    Text("Wallet History")
 }
 .disabled(viewModel.isSyncing)
 ````
 Pretty neat huh? well... no. This is a navigation bug ensured, because whenever the Synchronizer started to sync, the navigation stack would reset because the navigation link tied to the button was being disabled. 
 
Our app has the particular thing of being "passively" (it actually polls) receiving updates from the blockchain and having to react to them while the user is minding its own business. It could be the case that the user has the same seed in many different wallets and uses them depending on context (which is a not-so-good-practice, but it happens). So it could be the case that the user tried to use ZecWallet and sent zec. Then went back to Secant wallet minutes later and tried to send zec to another user he interacted originally on this app and then the blockchain updates with changes originated from the same keys but from another app, causing a balance change and disrupting a flow that needed certain conditions to be correctly shown like, the send flow, that depends on the wallet having a positive verified balance. 

If the navigation depends directly on UI components triggering the NavigationLink, the updates on the main State will trigger a navigation stack reset because the render engine understands the View Hierarchy branch has to be pruned since is not longer valid according to the State it is reading from the main view. 

So, done with the constitutional preamble 😜. Even though SwiftUI allows you to retroactively evaluate the navigation stack (mostly inadvertently), a safe and predictable Navigation state is the combination of user intent + application state at the time of the intent.
My main question is how can we express that in a way that is clear and easy to read (and debug) for developers and that doesn't leverage assumptions (like modifier evaluations) that could break by apple moving stuff around between releases, and that also is clearly assertive on code.

Reply:
> Then went back to Secant wallet minutes later and tried to send zec to another user he interacted originally on this app and then the blockchain updates with changes originated from the same keys but from another app, causing a balance change and disrupting a flow that needed certain conditions to be correctly shown like, the send flow, that depends on the wallet having a positive verified balance.
> 
> If the navigation depends directly on UI components triggering the NavigationLink, the updates on the main State will trigger a navigation stack reset because the render engine understands the View Hierarchy branch has to be pruned since is not longer valid according to the State it is reading from the main view.

This is a really intersting edge case! Would be good to quickly walk through it on another channel, but I think I understand what you mean. How we'd handle it depends on the desired UX for when this does happen, but in general I think things like this solvable in that the "on/off" binding does not have to corrospond to something in the state being completely on or off.

For example suppose we don't want to randomly pop people back based on something like `isSyncing` we can pull this into the state of the view itself, and the `NavigationLink` could be based on `isSyncing || isCurrentlySending`.

I'm not sure if that clarifies anything 😅 but let's catch up offline if it makes sense.

Reply:
Yes, probably the best thing is to actually resolve navigation programmatically when the user navigates and then resolve what happens when the child (e.g: send flow) can't operate with the current wallet state.